lib/cylc/network/client.py

```diff
@@ -239,11 +241,18 @@ def __new__(cls, suite, owner=None, host=None, port=None, timeout=None):
     @staticmethod
     def get_header():
         """Return "header" data to attach to each request for traceability."""
-        CYLC_EXE = os.path.join(os.environ['CYLC_DIR'], 'bin', '')
         cmd = sys.argv[0]
 
-        if cmd.startswith(CYLC_EXE):
-            cmd = cmd.replace(CYLC_EXE, '')
+        cylc_executable_location = which("cylc")
+        if cylc_executable_location:
+            cylc_bin_dir = os.path.abspath(
+                os.path.join(cylc_executable_location, os.pardir)
+            )
+            if not cylc_bin_dir.endswith("/"):
+                cylc_bin_dir = f"{cylc_bin_dir}/"
+
+            if cmd.startswith(cylc_bin_dir):
+                cmd = cmd.replace(cylc_bin_dir, '')
```

Comment:
I am probably missing the point of the logic here. Are we just saying that if `cmd` is `/path/to/cylc`, it is replaced with `cylc`?

Reply:
That's what I understood too. Only `cylc` commands, do the same as `basename` in Shell I think?

Reply:
`os.path.basename`?

Reply:
That one too. I was only a bit scared to change this as I am not familiar with this part of the code yet (i.e. not sure what is using this value, and how to test if I change it).

Using `os.path.basename`, I am not sure if the `cmd` isn't something like `/opt/cylc/bin/cylc run --no-detach --verbose five`... in which case the `.replace` would have to stay, in order to keep the arguments (also not sure if basename would work with arguments).

Maybe we can leave it as-is for now or open another issue to see how to improve or document this.

Reply:
Not possible looking at the code. It really can only be a simple `cylc` command.